packages/commerce-sdk-react/src/query-options.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {UseQueryOptions} from '@tanstack/react-query'
+
+export const getQueryOptions = <TQueryFnData, TError>(
```

Comment:
1. If we decide to provide a query configuration object to all individual queries, we should document this because this is not something that React Query does out of the box.

2. Why we are only using the query configuration object in the `commerce-sdk-react/src/hooks/ShopperProducts/query.ts` but not using this query configuration object in the rest of the hook queries of the library e.g. `commerce-sdk-react/src/hooks/ShopperSearch/query.ts`?

3. Observation. The `onError` function is something that users are going to override often when running mutations to invalidate cache entries and/or when implementing Optimistic Updates to implement rollbacks.

4. I recommend reading [TkDodo's React Query Error Handling post ](https://tkdodo.eu/blog/react-query-error-handling)
Developers would likely want to display errors to shoppers in a friendly way like using a
`useErrorBoundary` to display a fallback UI error or by using a toast.
It looks like this is the wrong place to display an error since [the onError callback on useQuery is called for every Observer](https://tkdodo.eu/blog/react-query-error-handling#:~:text=the%20onError%20callback%20on%20useQuery%20is%20called%20for%20every%20Observer).
If we want to only console log once per request it looks like a better approach is to handle this in the QueryClient configuration.

Reply:
About point 2 above: yup, I'll need to repeat the changes for the other hooks too.